src/Mutation/FilteredCollection.php

```diff
@@ -47,6 +47,46 @@ public function set(array $data): void
         }
     }
 
+    public function getIterator()
+    {
+        return new \RecursiveIteratorIterator($this->getRecursiveArrayIterator());
+    }
+
+    public function getRecursiveArrayIterator(): \RecursiveArrayIterator
+    {
+        return new class($this->arguments) extends \RecursiveArrayIterator {
+            public function hasChildren(): bool
+            {
+                $current = $this->current();
+                if ($current instanceof Item) {
+                    return false;
+                }
+
+                if (is_array($current)) {
+                    return true;
+                }
+
+                return $current->hasChildren();
+            }
+
+            public function getChildren()
+            {
+                return parent::current()->getRecursiveArrayIterator();
```

Comment:
We could use something like:

```
return parent::current()->getIterator()->getInnerIterator()
```

With this approach we will have just one public method to obtain an iterator.